### PR TITLE
add cleaner map

### PR DESCRIPTION
**Pull request by omar2205**

### CODE REVIEW

File: app/islands/Map.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          zoom={7}
+        >
+          <L.TileLayer
+            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+            attribution="© OpenStreetMap"
+            maxZoom="19"
+          />
+        </L.Map>
+      ) : (
+        <span></span>
```
Comment (omar2205) [inline app/islands/Map.tsx:37]: To use a null/undefined instead of an empty span, replace the fragments (<></>, L16+L39) with a div element.

**Comment by w3cj** [comment]
Thanks for this! Instead of merging, I am going to re-implement on stream so everyone can see how it works. I'll show your PR when I work on it.

**Comment by omar2205** [comment]
@w3cj, To get access to the map and do fun stuff like `locate()` and `panTo()`, I couldn't cleanly get access to it. However, if you made a ref to L.Map, there is `mapRef.current.state.map` Which gives you access to the map element.

I tried this, if I did `console.log(map.current.state.map)` map is there, but if I used setMap, it still is null.
```ts
  const [map, setMap] = useState<L.Map|null>(null)
  const mapRef = useRef(null)
  useEffect(() => {
    if ('state' in mapRef['current'])
      setMap(map.current.state.map)
  }, [mapRef])
```

**Comment by w3cj** [comment]
Just leaving a note on this, it looks like the library passes `leafletMap` as a prop to every child of the map, so that is probably the easiest way to get access to it.

https://github.com/Benjythebee/preact-leaflet-typescript/blob/master/src/Map.tsx#L107

Some pseudo code (haven't tested yet):

```jsx
const MapSetter = ({ setMap, leafletMap }) => {
  useEffect(() => {
    setMap(leafletMap);
  }, [leafletMap])
  return null;
}

const Page = () => {
  const [map, setMap] = useState(null);
  return (
    <Map>
      <MapSetter setMap={setMap} />
    </Map>
  );
}
```

**Pull request closed without merging**